docs/developers/release.md

```diff
@@ -4,10 +4,47 @@ This guide documents `napari`'s release process.
 Currently, it only handles distribution, but as the project matures,
 it will include generating release notes, documentation, etc.
 
-This is mainly meant for the core developers who will actually be performing the release.
-They will need to have a [PyPI](https://pypi.org) account with upload permissions to the `napari` package.
+# Timeline
+New versions of `napari` will be released every two months. The first release candidate will be available one week prior to release for testing purposes. Multiple release candidates may become available during the week prior to release.
```

Comment:
Do we want to actually say that the rc will become available first Tuesday of the month or no?

Reply:
Good point! Will add when on a computer